### PR TITLE
Add docker support

### PR DESCRIPTION
**Pull request by fentas**
image

### CODE REVIEW

**Comment by fentas** [comment]
Thanks for making this open-source! :100:

**Comment by hodgerpodger** [comment]
wow this is amazing!  thanks for contributing this.  I think I will merge this as is, so don't mind the below too much.  

I'm wondering how would I modify/add image or README so that I can perhaps optionally enter the docker image as a a shell, instead of directly executing the script.  Is it perhaps something like `docker run <some argument to overwrite ENTRYPOINT>`?

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+# install requirements
+COPY ./requirements.txt .
+RUN pip3 install -r requirements.txt
+
+# copy rest
+COPY . .
+
+# executable
+ENTRYPOINT [ "/staketaxcsv/docker_entrypoint.sh" ]
```
Comment (dting) [inline Dockerfile:32]: nit: missing new line?

File: docker_entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+fi
+shift
+
+# shortcut for docker volume binding
+[ ! -d /reports ] || [ -L src/_reports ] || {
+  rm -r src/_reports
+  ln -s /reports src/_reports
+}
+
+exec python3 "${script}" "${@}"
```
Comment (dting) [inline docker_entrypoint.sh:23]: nit: missing new line?

File: Dockerfile (hunk 1 above)
Comment (fentas) [reply Dockerfile:32]: Guess taste?
I like it without but feel free to add.

File: docker_entrypoint.sh (hunk 2 above)
Comment (fentas) [reply docker_entrypoint.sh:23]: same as above.
code wise it is not needed.

**Comment by fentas** [comment]
> wow this is amazing! thanks for contributing this. I think I will merge this as is, so don't mind the below too much.
> 
> I'm wondering how would I modify/add image or README so that I can perhaps optionally enter the docker image as a a shell, instead of directly executing the script. Is it perhaps something like `docker run <some argument to overwrite ENTRYPOINT>`?

ehm yeah it's plain docker
```
docker run --it --entrypoint /bin/sh staketaxcsv
```

**Comment by fentas** [comment]
https://github.com/hodgerpodger/staketaxcsv/pull/15/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R45-R49

**Comment by fentas** [comment]
Ok made the docker-entrypoint more classic. Also added a help. See

https://github.com/hodgerpodger/staketaxcsv/pull/15/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R45-R49

File: Dockerfile (hunk 1 above)
Comment (dting) [reply Dockerfile]: I think more than taste: 

https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

Github also surfaces this as "No newline at end of file" warning in the "no entry" icon when looking at file diffs:

https://github.com/hodgerpodger/staketaxcsv/pull/15/files

**Pull request merged**